schemas/dependencies/jira-board-1.yml

```diff
@@ -31,6 +31,8 @@ properties:
         type: string
     required:
     - workspace
+  issueResolveState:
```

Comment:
Last I checked there were only small no of distinct values ( less than 5). As this field is optional, I'm not planning on adding any constraints right now. Making this field enum with strict values would mean updating schemas each time a tenant comes with a different value and acts as a blocker. 

We can always comeback and make this an enum(add unique entries as enum) as that change would be backwards compatible.

Reply:
> As this field is optional, I'm not planning on adding any constraints right now

The field being optional doesn't prevent the addition of constraints? Or are you just suggesting it might not get much use initially?

> Making this field enum with strict values would mean updating schemas each time a tenant comes with a different value and acts as a blocker.

You're much closer to this than me, but are these values categories defined by Jira (small number of values), or are they user-defined values? If the former, then maybe having an enum will be less work than dealing with errors generated from typos and such. If the latter, then I agree that we don't want to be maintaining that list.

I agree we can always do it later, but never hurts to handle it early if it's low effort and we have reason to believe the number of values will stay small. I'll approve and let you decide the best path forward.

Reply:
> > As this field is optional, I'm not planning on adding any constraints right now
> 
> The field being optional doesn't prevent the addition of constraints? Or are you just suggesting it might not get much use initially?

I meant the former. I'm pretty sure json-schema should allow enum for optional fields.
> > Making this field enum with strict values would mean updating schemas each time a tenant comes with a different value and acts as a blocker.
> 
> You're much closer to this than me, but are these values categories defined by Jira (small number of values), or are they user-defined values? If the former, then maybe having an enum will be less work than dealing with errors generated from typos and such. If the latter, then I agree that we don't want to be maintaining that list.

From what I could gather from sources, a user can define any resolve status such as `Foo`. I guess Jira probably also comes up with its own resolution state but from the sources it looks like user can define any arbitrary resolution status.

> I agree we can always do it later, but never hurts to handle it early if it's low effort and we have reason to believe the number of values will stay small. I'll approve and let you decide the best path forward.